shakenfist/daemons/queues.py

```diff
@@ -95,6 +96,14 @@ def handle(jobname, workitem):
                 except Exception as e:
                     util.ignore_exception(daemon.process_name('queues'), e)
 
+            elif isinstance(task, SnapshotInstanceTask):
+                snap_uuid = task.snap_uuid
+                instance = virt.Instance.from_db(instance_uuid)
+                instance.snapshot(all=all, snapshot_uuid=snap_uuid)
```

Comment:
Probably should check that the instance is not deleted here? Also, check that the instance exists at all.

Reply:
You're right. This is checked when the task is created, but not again when it's time to handle it and things could have changed since them.